frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { useAuthenticatedUser } from '@ifixit/auth-sdk';
+import { ProductSearchHit } from '@models/product-list';
+import { computeDiscountPercentage } from '@helpers/commerce-helpers';
+
+export type ProductSearchHitPricing = {
+   price: number;
+   compareAtPrice: number | undefined;
+   isDiscounted: boolean;
+   percentage: number;
+};
+export function useProductSearchHitPricing(
+   product: ProductSearchHit
+): ProductSearchHitPricing {
+   const user = useAuthenticatedUser();
+
+   const proTierPrice = React.useMemo(() => {
+      const proTier = user.data?.discountTier ?? null;
+      if (proTier) {
+         const priceString = product.price_tiers?.[proTier];
+         return priceString == null ? null : parseFloat(priceString);
+      }
+      return null;
+   }, [user.data?.discountTier, product.price_tiers]);
+
+   const isProUser = proTierPrice != null;
+
+   const price = proTierPrice ?? product.price_float;
+   const compareAtPrice = isProUser
+      ? product.compare_at_price ?? product.price_float
+      : product.compare_at_price;
```

Comment:
```suggestion
   const price = proTierPrice ?? product.price_float;
   const compareAtPrice = product.compare_at_price ?? product.price_float;
```

If we just always fallback to the regular price, then `isDiscounted` will still protect us from showing the same price twice.

Reply:
lgtm 👍